mkosi

```diff
@@ -2313,14 +2314,34 @@ def detect_distribution():
 
     id = None
     version_id = None
+    version_codename = None
+    extracted_codename = None
 
     for ln in f:
         if ln.startswith("ID="):
             id = ln[3:].strip()
         if ln.startswith("VERSION_ID="):
             version_id = ln[11:].strip()
+        if ln.startswith("VERSION_CODENAME="):
+            version_codename = ln[17:].strip()
+        if ln.startswith("VERSION="):
+            # extract Debian release codename
+            version_str = ln[8:].strip()
+            debian_codename_re = r'\((.*?)\)'
+
+            codename_list = re.findall(debian_codename_re, version_str)
+            if len(codename_list) == 1:
+                extracted_codename = codename_list[0]
```

Comment:
Does this actually work? I see:
```
VERSION="17.04 (Zesty Zapus)"
VERSION="17.10 (Artful Aardvark)"
```
and my debian-unstable container doens't even have a `VERSION` field.

But even if the field is present, the result will be `"Zesty Zapus"`, but we need `zesty`, no?

Reply:
> Does this actually work?

Yes, for Debian. Tested on Stretch (stable) and Buster (unstable). It's true that Buster install doesn't have the ```VERSION``` field (and ```VERSION_ID``` is also missing) in the ```/etc/os-release``` but in that case ```mkosi``` already falls back into ```unstable``` - see ```load_args``` function.

Regarding Ubuntu, I have not tested but this PR shouldn't change anything on Canonical's distribution. Should I also have a look there?